spec/datadog/core/telemetry/v1/app_started_spec.rb

```diff
@@ -122,4 +122,39 @@
       end
     end
   end
+
+  describe '#to_h' do
+    subject(:to_h) { app_started.to_h }
+
+    context 'when attributes are all nil' do
+      let(:additional_payload) { nil }
+      let(:configuration) { nil }
+      let(:dependencies) { nil }
+      let(:integrations) { nil }
+      it { is_expected.to eq({}) }
+    end
+
+    context 'when attributes are all defined' do
+      let(:additional_payload) { { 'tracing.enabled' => true, 'profiling.enabled' => false } }
+      let(:configuration) { { DD_AGENT_HOST: 'localhost', DD_TRACE_SAMPLE_RATE: '1' } }
+      let(:dependencies) { [Datadog::Core::Telemetry::V1::Dependency.new(name: 'pg')] }
+      let(:integrations) { [Datadog::Core::Telemetry::V1::Integration.new(name: 'pg', enabled: true)] }
+
+      before do
+        allow(dependencies[0]).to receive(:to_h).and_return({ name: 'pg' })
+        allow(integrations[0]).to receive(:to_h).and_return({ name: 'pg', enabled: true })
+      end
```

Comment:
Is this still needed? It looks to me that both of these get `#to_h` methods in this PR, so is there any difference between the actual implementations and the mock responses we're setting up here?

(Also, this happens on other files, so if you do change something based on my feedback, do a quick pass over all the files in the PR and also update other places where this is being used)